pkg/security/security_profile/profile/profile.go

```diff
@@ -157,21 +173,48 @@ func (p *SecurityProfile) SendStats(client statsd.ClientInterface) error {
 }
 
 // ToSecurityProfileMessage returns a SecurityProfileMessage filled with the content of the current Security Profile
-func (p *SecurityProfile) ToSecurityProfileMessage(timeResolver *timeResolver.Resolver, cfg *config.RuntimeSecurityConfig) *api.SecurityProfileMessage {
+func (p *SecurityProfile) ToSecurityProfileMessage() *api.SecurityProfileMessage {
+	// construct the list of image tags for this profile
+	imageTags := ""
+	for key := range p.versionContexts {
+		if imageTags != "" {
+			imageTags = imageTags + ","
+		}
+		imageTags = imageTags + key
+	}
+
 	msg := &api.SecurityProfileMessage{
 		LoadedInKernel:          p.loadedInKernel,
-		LoadedInKernelTimestamp: timeResolver.ResolveMonotonicTimestamp(p.loadedNano).String(),
+		LoadedInKernelTimestamp: p.timeResolver.ResolveMonotonicTimestamp(p.loadedNano).String(),
 		Selector: &api.WorkloadSelectorMessage{
 			Name: p.selector.Image,
-			Tag:  p.selector.Tag,
+			Tag:  imageTags,
 		},
 		ProfileCookie: p.profileCookie,
-		Version:       p.Version,
 		Metadata: &api.MetadataMessage{
 			Name: p.Metadata.Name,
 		},
-		Tags: p.Tags,
+		ProfileGlobalState: p.GetGlobalState().String(),
+		ProfileContexts:    make(map[string]*api.ProfileContextMessage),
+	}
+	p.versionContextsLock.Lock()
```

Comment:
Should this lock be moved at beginning of the function (before the imageTags string is constructed)?

Reply:
Suggestion applied on last commit, thank you!